c/extras/aggregator.cc

```diff
@@ -389,15 +389,17 @@ double* Aggregator::generate_pmatrix(DataTablePtr& dt_exemplars) {
 void Aggregator::project_row(DataTablePtr& dt_exemplars, double* r, int32_t row_id, double* pmatrix) {
   std::memset(r, 0, static_cast<size_t>(max_dimensions) * sizeof(double));
 
-  for (int32_t i = 0; i < (dt_exemplars->ncols) * max_dimensions; ++i) {
-    int col_id = i / max_dimensions;
-    if (dt_exemplars->columns[col_id]->stype() == SType::FLOAT64) {
-      RealColumn<double>* c = static_cast<RealColumn<double>*> (dt_exemplars->columns[col_id]);
-      const double* d_c = static_cast<const double*>(dt_exemplars->columns[col_id]->data());
-
+  for (int32_t i = 0; i < dt_exemplars->ncols; ++i) {
+    if (dt_exemplars->columns[i]->stype() == SType::FLOAT64) {
+      const double* d_c = static_cast<const double*>(dt_exemplars->columns[i]->data());
```

Comment:
A minor note: whenever you have `TYPE var = static_cast<TYPE>(...)` it's better to declare the variable as `auto`. That way you repeat yourself less, and the line is shorter too:
```C++
auto d_c = static_cast<const double*>(dt_exemplars->columns[i]->data());
// vs
const double* d_c = static_cast<const double*>(dt_exemplars->columns[i]->data());
``` 

Also, in this particular case you could have declared `d_c` first, and then
```C++
const double* d_c = c->elements_r();
```

Reply:
Sure, fixing it now.

Reply:
As for your second comment, I guess you meant 
>you could have declared `c` first
?

If so, it could be the case that this particular value is missing and we don't really need to declare `c` at all. Probably in most of the cases the value is not missing, so what you say makes perfect sense.